nav2_collision_monitor/src/polygon.cpp

```diff
@@ -168,6 +169,37 @@ bool Polygon::isShapeSet()
   return true;
 }
 
+bool Polygon::isUsingPolygonVelocitySelector()
+{
+  return !polygons_velocity_.empty();
+}
+
+void Polygon::updatePolygonByVelocity(const Velocity & cmd_vel_in)
+{
+  for (auto & polygon_velocity : polygons_velocity_) {
+
+    if (polygon_velocity->isInRange(cmd_vel_in)) {
+      // Set the polygon that is within the speed range
+      poly_ = polygon_velocity->getPolygon();
+
+      // Update visualization polygon
+      polygon_.polygon.points.clear();
+      for (const Point & p : poly_) {
+        geometry_msgs::msg::Point32 p_s;
+        p_s.x = p.x;
+        p_s.y = p.y;
+        // p_s.z will remain 0.0
+        polygon_.polygon.points.push_back(p_s);
+      }
+
+      return;
+    }
+  }
+  RCLCPP_WARN(
```

Comment:
Make this a throttled warning every few seconds. This will blast the terminal at controller rate which would get really annoying